lib/elixir/lib/task/supervised.ex

```diff
@@ -343,22 +378,23 @@ defmodule Task.Supervised do
     # process waits, this process dies with the same reason.
     receive do
       {^parent_pid, monitor_ref} ->
-        stream_monitor_loop(parent_pid, parent_ref, mfa, spawn, monitor_ref, _counters = %{})
+        stream_monitor_loop(parent_pid, parent_ref, mfa, spawn, monitor_ref, _counters = %{}, timeout)
       {:DOWN, ^parent_ref, _, _, reason} ->
         exit(reason)
     end
   end
 
-  defp stream_monitor_loop(parent_pid, parent_ref, mfa, spawn, monitor_ref, counters) do
+  defp stream_monitor_loop(parent_pid, parent_ref, mfa, spawn, monitor_ref, counters, timeout) do
     receive do
       # The parent process is telling us to spawn a new task to process
       # "value". We spawn it and notify the parent about its pid.
       {:spawn, counter, value} ->
         {type, pid} = spawn.(parent_pid, normalize_mfa_with_arg(mfa, value))
         ref = Process.monitor(pid)
-        send(parent_pid, {:spawned, {monitor_ref, counter}, pid})
+        timer_ref = Process.send_after(parent_pid, {:timed_out, {monitor_ref, counter}}, timeout)
```

Comment:
When starting a timer for a different process it is not possible to guarantee that the other process receives the timer reference because it is always possible for the process to get an exit signal in between starting the monitor and sending the timer reference. Therefore to prevent timer leaks we would require the different process to always go down, i.e. using links, monitors or supervisors. Unfortunately can't guarantee this here.

It could happen in the following situation:
* `on_timeout: :exit` and catches exit when stream enumerated
* Monitor process gets exit signal that makes it exit between `Process.send_after` and `send`
* Caller is trapping exits when it receives the exit signal from monitor process (need not be when spawning monitor process but still possible if `:kill`)